pdcupdater/handlers/modules.py

```diff
@@ -0,0 +1,267 @@
+        with TmpDir(prefix="pdcupdater-") as tmpdir, PushPopD(tmpdir), \
+                open(os.devnull, "w") as devnull:
+            m = self.scmurl_re.match(scmurl)
+            if not m:
+                log.error("Can't parse SCM URL: {}".format(scmurl))
+                raise RuntimeError("Can't parse SCM URL: {}".format(scmurl))
+            giturl = m.group('giturl')
+            repopath = m.group('repopath')
+            modpath = m.group('modpath')
+            revision = m.group('revision')
+            while repopath.endswith("/"):
+                repopath = repopath[:-1]
+            modname = repopath.rsplit("/", 1)[1]
+            if modname.endswith(".git"):
+                modname = modname[:-4]
+
+            log.debug("Cloning {}".format(giturl))
+            check_call(["git", "clone", "-n", giturl, modname],
+                       stdout=devnull, stderr=STDOUT)
+            os.chdir(modname)
+
+            log.debug("Checking out revision {}".format(revision))
+            check_call(["git", "reset", "--hard", revision],
+                       stdout=devnull, stderr=STDOUT)
+
+            mmd_yaml = modname + ".yaml"
+            if modpath:
+                mmd_yaml = "/".join((modpath, mmd_yaml))
+            log.debug("Reading/parsing {}".format(mmd_yaml))
+            mmd = modulemd.ModuleMetadata()
+            mmd.load(mmd_yaml)
+
+            return mmd
+
+    def create_unreleased_variant(self, pdc, body):
+        """Creates an UnreleasedVariant for a module in PDC. Checks out the
+        module metadata from the supplied SCM repository (currently only
+        anonymous GIT is supported)."""
+
+        scmurl = body['scmurl']
+        mmd = self.get_mmd_from_scm(scmurl)
+
+        runtime_deps = []
+        for dep, ver in mmd.requires.items():
+            if ver is not None:
+                runtime_deps.append("{} >= {}".format(dep, ver))
+            else:
+                runtime_deps.append(dep)
+
+        build_deps = []
+        for dep, ver in mmd.buildrequires.items():
+            if ver is not None:
+                build_deps.append("{} >= {}".format(dep, ver))
+            else:
+                build_deps.append(dep)
+
+        koji_tag = body['koji_tag']
+        variant_uid = variant_name = body['module_uid']
+        variant_id = variant_uid.lower()
+        variant_version = body['module_version']
+        variant_release = body['module_release']
+
+        unreleased_variant = pdc['unreleasedvariants']._({
+            'variant_id': variant_id,
+            'variant_uid': variant_uid,
+            'variant_name': variant_name,
+            'variant_version': variant_version,
+            'variant_release': variant_release,
+            'variant_type': 'module',
+            'koji_tag': koji_tag,
+            'runtime_deps': runtime_deps,
+            'build_deps': build_deps,
+        })
+
+        return unreleased_variant
+
+    def get_or_create_unreleased_variant(self, pdc, body):
```

Comment:
I had trouble reading and understanding this function when I first encountered it.

Can you add some inline comments to it explaining the thought?